config/toml.go

```diff
@@ -322,20 +323,8 @@ func ResetTestRoot(testName string) *Config {
 }
 
 func ResetTestRootWithChainID(testName string, chainID string) *Config {
-	rootDir := os.ExpandEnv("$HOME/.tendermint_test")
-	rootDir = filepath.Join(rootDir, testName)
-	// Remove ~/.tendermint_test_bak
-	if cmn.FileExists(rootDir + "_bak") {
-		if err := os.RemoveAll(rootDir + "_bak"); err != nil {
-			cmn.PanicSanity(err.Error())
-		}
-	}
-	// Move ~/.tendermint_test to ~/.tendermint_test_bak
-	if cmn.FileExists(rootDir) {
-		if err := os.Rename(rootDir, rootDir+"_bak"); err != nil {
-			cmn.PanicSanity(err.Error())
-		}
-	}
+	// create a unique, concurrency-safe test directory under $HOME/.tendermint_test/
+	rootDir := mustCreateTestRootDir(testName, chainID)
 	// Create new dir
 	if err := cmn.EnsureDir(rootDir, 0700); err != nil {
```

Comment:
note you create directory twice now. 1) ioutil.Tempdir 2) cmn.EnsureDir here

Reply:
I think you're right, the `EnsureDir` following the `mustCreateTestRootDir` is basically a no-op. Will remove then, thanks.